dissect/target/plugins/filesystem/yara.py

```diff
@@ -16,48 +24,128 @@
         ("digest", "digest"),
         ("string", "rule"),
         ("string[]", "tags"),
+        ("string", "namespace"),
     ],
 )
 
+MAX_SCAN_SIZE = 10 * 1024 * 1024
 
-class YaraPlugin(Plugin):
-    """Plugin to scan files against a local YARA rules file."""
 
-    DEFAULT_MAX_SIZE = 10 * 1024 * 1024
+class YaraPlugin(InternalPlugin):
+    """Plugin to scan files against a local YARA rules file."""
 
     def check_compatible(self) -> None:
-        pass
+        if not HAS_YARA:
+            raise UnsupportedPluginError("Please install 'yara-python' to use the yara plugin.")
+
+    def yara(
+        self, rules: str | list[str | Path], path: str = "/", max_size: int = MAX_SCAN_SIZE, check_rules: bool = False
+    ) -> Iterator[YaraMatchRecord]:
+        """Scan files inside the target up to a given maximum size with YARA rule file(s)."""
+        rules_path = rules
+
+        if isinstance(rules, str):
+            rules = rules.split(",")
 
-    @arg("--rule-files", "-r", type=Path, nargs="+", required=True, help="path to YARA rule file")
-    @arg("--scan-path", default="/", help="path to recursively scan")
-    @arg("--max-size", "-m", default=DEFAULT_MAX_SIZE, help="maximum file size in bytes to scan")
-    @export(record=YaraMatchRecord)
-    def yara(self, rule_files, scan_path="/", max_size=DEFAULT_MAX_SIZE):
-        """Scan files up to a given maximum size with a local YARA rule file.
+        rules = process_rules(rules, check_rules)
 
-        Example:
-            target-query <TARGET> -f yara --rule-file /path/to/yara_sigs.rule
-        """
+        if not rules:
+            self.target.log.error("No working rules found in %s.", rules_path)
+            return set(())
```

Comment:
If you have a yield anywhere in your code, you can return from anywhere and it will exit the generator:
```python
In [1]: def a():
   ...:     for i in range(5):
   ...:         yield i
   ...:         if i == 3:
   ...:             return
   ...: 

In [2]: list(a())
Out[2]: [0, 1, 2, 3]
```